### DISCUSSION TITLE
Stuck at shell on 3.5in GPIO touchscreen

### DISCUSSION THREAD

**Discussion by red4dj** [guysoft/FullPageOS#564] [Q&A]
#### What were you doing?

Running FullScreenOS on 3.5in GPIO touchscreen after following driver install https://www.waveshare.com/wiki/3.5inch_RPi_LCD_(A) or http://www.lcdwiki.com/3.5inch_RPi_Display

#### What did you expect to happen?

Browser to launch

#### What happened instead?

Stuck at terminal

#### Was there an error message displayed? What did it say?

Running start_gui script causes
```
matchbox: can't open display! check your DISPLAY variable.
Error: Can't open display: (null)
```

#### Version of FullPageOS?

0.13.0 stable 2022-03-07

Identical to issue #120 but fix provided by @xbeaudouin and referenced in wiki no longer applicable as the conf.txt no longer contains this line after running driver bash script. Tried similar fixes based on what's available to no avail.

**Comment by guysoft** [comment]
Converted to a discussion, since this is not something planned to be supported.
They have driver install instructions in the link you provided - did you try that?

**Reply by craftcm** [reply]
I'm actually having the same problem with the 3.5 in TFT screen connected up via GPIO. I'm running the Lite OS with no display connected via HDMI - only the 3.5 in TFT screen. Is this not a supported use case?

**Reply by red4dj** [reply]
Have used multiple drivers from multiple groups. Also went through a bunch of my own troubleshooting. I understand if this project doesn't plan to support it, however this does seem like a valid use case for if I want to display a dashboard webpage on a small, interactive screen (think embedded in a wall).

Is this something that should be brought up to matchbox?

**Reply by guysoft** [reply]
I think the issue is that a lot of those TFT screens have a custom kernel driver/device tree, and use the SPI connection. And that means you need to have the correct device tree, which is rarely maintained in newer versions. Not sure how to help per-screen. It depends on the brand etc. You might need to set up your own device tree (which is not fun from experience, its a nice rabbit hole if you like to tinker though).

**Reply by ThatCameraOperator** [reply]
Was this fixed for the 3.5" displays?
If I set up an RPi3 with the 5" screen connected via HDMI, FullPageOS guides all have me with a working Tally Light screen for camera operators from the http tally output of vMix.
But if I try with the 3.5" LCD, using the drivers mentioned above or the 'goodtft' version, it displays the splash screen then kicks back out to the terminal display.
(Wryly) It's disappointing because I have two RPi3's, two 3.5" LCD's,... and Two Camera Operators for an upcoming gig. ;)
The smaller size of the 3.5" screen results in a smaller package once in a case and sitting on the camera's cold-shoe.
